kubernetes/assets/monitors/monitor_cluster_memory.json

```diff
@@ -2,7 +2,7 @@
 {
 	"name": "[kubernetes] Monitor Kubernetes Cluster Memory Usage",
 	"type": "query alert",
-	"query": "avg(last_5m):sum:kubernetes.memory.requests{*} by {kubernetes_cluster} * 100 / sum:kubernetes.memory.capacity{*} by {kubernetes_cluster} > 90",
+	"query": "avg(last_5m):avg:kubernetes.memory.usage_pct{*} by {kubernetes_cluster} > 0.9",
 	"message": "Memory usage in reaching maximum capacity in {{kubernetes_cluster.name}}",
```

Comment:
could we recommend actionable next steps?

Reply:
Will follow up on next steps for all monitors with the containers team.

Reply:
This is a cluster capacity issue. In most cases adding more workers is the answer. Or, if the cluster has reached its max capacity, then adding new clusters would be an option. Sometimes operators maybe able to perform some cleanup for unneeded deployments/statefults/etc that are eating up memory. Using cluster-autoscaler is a way to dynamically increase your cluster capacity but it is more relevant when the clusters are running in a cloud. One other things folks can do is use bigger nodes but again this depends on how they are running their kube worker nodes.

Also, I'm thinking whether using this metrics could be slightly misleading here `kubernetes.memory.usage_pct` (I think we may have recommended it but depending upon the intent of this monitor, it may or may not be the correct one to use). Essentially, it is looking at the memory usage of pods (vs memory available on nodes, which is what dictates cluster capacity). If we have only 10 pods on a cluster with 1000s of nodes, it is possible to hit 90% of usage because we are only looking at the memory usage of the pods vs the memory available on the cluster, which is a lot more. Does that make sense? Or, am I missing something here?

Reply:
hmm ok - will rework this monitor in a separate review. Removing it for now in the interests of time.